Implicit transitive 'csstype' dependency

<!-- Provide a general summary of the issue in the Title above -->


<!--
  Thank you very much for contributing to Material-UI by creating an issue! ❤️
  To avoid duplicate issues we ask you to check off the following list.
-->

<!-- Checked checkbox should look like this: [x] -->

- [x] The issue is present in the latest release.
- [x] I have searched the [issues](https://github.com/mui-org/material-ui/issues) of this repository and believe that this is not a duplicate.

## Current Behavior 😯

Yarn build fails with TypeScript and Yarn PnP

```
../../.yarn/$$virtual/@material-ui-core-virtual-fc07fd2afd/0/cache/@material-ui-core-npm-5.0.0-alpha.14-30994eb5d9-8f221c1526.zip/node_modules/@material-ui/core/styles/experimentalStyled.d.ts:2:22 - error TS2307: Cannot find module 'csstype' or its corresponding type declarations.

2 import * as CSS from 'csstype';
```


## Your Environment 🌎

<!--
  Include as many relevant details about the environment with which you experienced the bug.
  If you encounter issues with TypeScript please include version and tsconfig.
-->

| Tech        | Version |
| ----------- | ------- |
| Material-UI | v5.0.0-alpha.14  |
| React       |   17      |
| Yarn     |      2   |
| TypeScript  |  4       |
| etc.        |         |

--- a/packages/material-ui/package.json
+++ b/packages/material-ui/package.json
@@ -68,6 +68,7 @@
     "@popperjs/core": "^2.4.4",
     "@types/react-transition-group": "^4.2.0",
     "clsx": "^1.0.4",
+    "csstype": "^3.0.2",
     "hoist-non-react-statics": "^3.3.2",
     "prop-types": "^15.7.2",
     "react-is": "^16.8.0",

